go-runtime/compile/schema.go

```diff
@@ -300,6 +314,57 @@ func extractTypeDeclsForNode(pctx *parseContext, node *ast.GenDecl) {
 	}
 }
 
+func topicNameFromCallExpr(node *ast.CallExpr) (string, *schema.Error) {
+	if len(node.Args) == 1 {
```

Comment:
I'd love to see if we could build some kind of pattern matching abstraction on top of the AST/type analyser, as this code we keep having to write is super repetitive. Maybe there's some existing package for it TBH, but something like this:

```go
name, err := ExtractFuncArg1[string](node, "ftl.RegisterTopic")
```

Whereby it matches the function, then can figure out the correct thing to do for literals of certain types such as `string`, `int`, etc. and  perhaps even other function calls.

Reply:
I ended up making one that just works for string literals (see `extractStringLiteralArg(...)`), and that helped some repeated code in the file